src/services/member/plugins/profile/repository.ts

```diff
@@ -33,12 +30,17 @@ const MemberProfileRepository = AppDataSource.getRepository(MemberProfile).exten
     return memberProfile;
   },
 
-  async getByMemberId(memberId: string, filter?: ProfileFilter): Promise<MemberProfile | null> {
+  async getByMemberId(
+    memberId: string,
+    filter?: {
+      visibility?: boolean;
+    },
+  ): Promise<MemberProfile | null> {
     if (!memberId) {
       throw new MemberNotFound();
     }
     const memberProfile = await this.findOne({
-      where: { member: { id: memberId }, ...filter },
+      where: { member: { id: memberId }, visibility: filter?.visibility },
```

Comment:
Would this need a default value? I guess it does not consider the prop if it is `undefined`, but not sure?

Reply:
It should safely send undefined when the visibility is not defined.